source/dub/compilers/compiler.d

```diff
@@ -115,19 +115,53 @@ interface Compiler {
 		enforce(status == 0, format("%s failed with exit code %s.", args[0], status));
 	}
 
+	/** Detect the frontend version of a compiler_binary
+	    Returns: 0 if the detection failed, otherwise the frontend version.
+		Note: This only works for DMD and LDC.
+	 */
+	private int frontendVersion(string compiler_binary)
+	{
+		import std.range : dropOne;
+
+		auto res = execute([compiler_binary, "--version"]);
+		if (res.status == 0) {
+			auto versionRange = res.output.find("DMD64 D Compiler v").splitter(".");
+			if (!versionRange.empty && !versionRange.dropOne.empty) {
+				return versionRange.dropOne.front.to!int;
+			}
+		}
+		return 0;
+	}
+
+	// DMD 2.076.0 and newer support passing files via stdin
+	private bool canUseStdin(string compiler_binary)
+	{
+		return frontendVersion(compiler_binary) >= 76 && compiler_binary.startsWith("dmd");
+	}
+
 	/// Compiles platform probe file with the specified compiler and parses its output.
 	protected final BuildPlatform probePlatform(string compiler_binary, string[] args, string arch_override)
 	{
 		import std.string : format;
-		import dub.compilers.utils : generatePlatformProbeFile, readPlatformProbe;
+		import dub.compilers.utils : generatePlatformProbeFile, readPlatformProbe, platformProbeFile;
 
-		auto fil = generatePlatformProbeFile();
+		bool canUseStdin = this.canUseStdin(compiler_binary);
+		string fil = canUseStdin ? generatePlatformProbeFile().toNativeString() : "-";
```

Comment:
Of course we could also remove the temp file after the exit of the current scope.